src/samplebuffer.h

```diff
@@ -0,0 +1,90 @@
+#ifndef SAMPLEBUFFER_H
+#define SAMPLEBUFFER_H
+
+#include "util/types.h" // CSAMPLE
+
+#include <algorithm> // std::swap
+
+// A sample buffer with properly aligned memory to enable SSE optimizations.
+//
+// No resize operation is provided intentionally for maximum efficiency!
+// If the size of an existing sample buffer needs to be altered after
+// construction this can simply be achieved by swapping the contents with
+// a temporary sample buffer that has been constructed with the appropriate
+// size:
+//
+//     SampleBuffer sampleBuffer(oldSize);
+//     ...
+//     SampleBuffer(newSize).swap(sampleBuffer);
+//     // Now the data in sampleBuffer is uninitialized and
+//     // sampleBuffer.size() == newSize
+//
+// No sample data is copied when swapping the contents of two samples buffers
+// for resizing!
+class SampleBuffer {
+    Q_DISABLE_COPY(SampleBuffer);
+
+  public:
+    typedef size_t size_type;
+    typedef CSAMPLE value_type;
```

Comment:
> Counter question: Why didn't you use a typedef if 'int' is essential for those loops to get optimized?

I don't think int is essential for those loops being optimized -- based on the source Daniel provided describing the MSVC loop vectorizer, unsigned int can cause problems if it does not match the natural architecture width but size_t will always be the full integer width of the architecture.

I think size_t is the ideal type to use since it semantically describes what the variable represents. Pretty much the only spots that this matters are index variables in for loops anyway and I already had to change all those in https://github.com/mixxxdj/mixxx/commit/0cdafa3cc0b5679f9d809740928e20844986564a
once #483 merged.